include/zenoh-pico/system/common.h

```diff
@@ -57,20 +57,28 @@ time_t z_time_elapsed_ms(z_time_t *time);
 time_t z_time_elapsed_s(z_time_t *time);
 
 /*------------------ Network ------------------*/
-_ZN_RESULT_DECLARE(_zn_socket_t, socket)
+int _zn_send_wbuf(_zn_link_t *link, const _z_wbuf_t *wbf);
+int _zn_recv_zbuf(_zn_link_t *link, _z_zbuf_t *zbf);
+int _zn_recv_exact_zbuf(_zn_link_t *link, _z_zbuf_t *zbf, size_t len);
 
 char *_zn_select_scout_iface(void);
-_zn_socket_result_t _zn_open_tx_session(const char *locator);
-void _zn_close_tx_session(_zn_socket_t sock);
 
-struct sockaddr_in *_zn_make_socket_address(const char *addr, int port);
-_zn_socket_result_t _zn_create_udp_socket(const char *addr, int port, int recv_timeout);
-
-int _zn_send_dgram_to(_zn_socket_t sock, const _z_wbuf_t *wbf, const struct sockaddr *dest, socklen_t salen);
-int _zn_recv_dgram_from(_zn_socket_t sock, _z_zbuf_t *zbf, struct sockaddr *from, socklen_t *salen);
-
-int _zn_send_wbuf(_zn_socket_t sock, const _z_wbuf_t *wbf);
-int _zn_recv_zbuf(_zn_socket_t sock, _z_zbuf_t *zbf);
-int _zn_recv_bytes(_zn_socket_t sock, uint8_t *buf, size_t len);
+// TCP
```

Comment:
I believe TCP and UDP (and any other) protocols should be compiled optionally through appropriate `#define` directives.

Reply:
Agreed